Documentation/MyFirstContribution

```diff
@@ -0,0 +1,584 @@
+
+----
+test_expect_success 'runs correctly with no args' '
+	rm -rf workbench upstream &&
+	test_create_repo upstream &&
+	(
+		cd upstream &&
+		git psuh
+	)
+'
+----
+
+Indicate that you've run everything you wanted by adding the following at the
+bottom of your script:
+
+----
+test_done
+----
+
+=== Running Locally
+
+Let's try and run locally:
+
+----
+make -j16
+cd t/ && prove t9999-psuh-codelab.sh
+----
+
+You can run the full test suite and ensure git-psuh didn't break anything:
+
+----
+cd t/
+prove -j16 --shuffle t[0-9]*.sh
+----
+
+(You can also do this with `make test` but `prove` can run concurrently.
+Shuffle randomizes the order the tests are run in, which makes them resilient
+against unwanted inter-test dependencies. `prove` also makes the output nicer.
+
+Go ahead and commit this change, as well.
+
+== Getting Ready to Share
+
+Before you send your patch off to be reviewed by the wide world, it's a good
+idea to run the Travis CI build and test suites against your new changes. You
+can do this manually or by using GitGitGadget, but either way, you're going to
+need to fork. First thing - make sure you have a GitHub account.
+
+=== Forking git/git on GitHub
+
+Head to the https://github.com/git/git[GitHub mirror] and look for the Fork
+button. Place your fork wherever you deem appropriate and create it.
+
+=== Uploading To Your Own Fork
+
+To upload your branch to your own fork, you'll need to add the new fork as a
+remote. You can use `git remote -v` to show the remotes you have added already.
+From your new fork's page on GitHub, you can press "Clone or download" to get
+the URL; then you need to run the following to add, replacing your own URL and
+remote name for the examples provided:
+
+----
+git remote add remotename git@github.com:remotename/git.git
+----
+
+Run `git remote -v` again and you should see the new remote showing up.
+`git fetch remotename` (with the real name of your remote replaced) in order to
+get ready to push.
+
+Next, double-check that you've been doing all your development in a new branch
+by running `git branch`. If you didn't, now is a good time to move your new
+commits to their own branch.
+
+As mentioned briefly at the beginning of this doc, we are basing our work on
+master, so go ahead and update as shown below, or using your preferred
+workflow.
+
+----
+git checkout master
+git pull -r
+git rebase master psuh
+----
+
+Finally, you're ready to push your new topic branch! (Due to our branch and
+command name choices, be careful when you type the command below.)
+
+----
+git push remotename psuh
+----
+
+Now you should be able to go and check out your newly created branch on GitHub.
+
+////
+TODO: The next few bullets describe testing and pushing your change with
+GitGitGadget. It may be useful to describe a workflow using git send-email as
+well.
+////
+
+=== Sending a PR to GitGitGadget
+
+GitGitGadget is a tool created by Johannes Schindelin to make life as a Git
```

Comment:
Personally, I'd prefer it if the reason why the tool exists (PRs are easy, sending patches via mail without letting mail change the formatting is not) was described here, not who did it ;-)